services/venice-router/src/main/java/com/linkedin/venice/router/RouterServer.java

```diff
@@ -852,20 +852,19 @@ public void stopInner() throws Exception {
      * correctly.
      */
 
+    LOGGER.info("Waiting to make sure all in-flight requests are drained");
     // Graceful shutdown: Wait till all the requests are drained
     try {
       RetryUtils.executeWithMaxAttempt(() -> {
-        if (RouterHttpRequestStats.hasInFlightRequests()) {
-          throw new VeniceException("There are still in-flight requests in router");
+        if (hasInFlightRequest()) {
```

Comment:
I think we should just make one call instead two:
```
long inFlightRequestCount = getInFlightRequestCount();
if (inFlightRequestCount > 0) {
    throw new VeniceException("There are still in-flight requests in router :" + inFlightRequestCount);
}
```
Two calls can return inconsistent results and we can completely remove function: `hasInFlightRequest()`.